src/main/io/vtx_tramp.c

```diff
@@ -0,0 +1,399 @@
+        trampRFPowerMax = trampRespBuffer[6]|(trampRespBuffer[7] << 8);
+        trampStatus = TRAMP_STATUS_ONLINE;
+        break;
+
+    case 'v':
+        trampCurFreq = trampRespBuffer[2]|(trampRespBuffer[3] << 8);
+        vtx58_Freq2Bandchan(trampCurFreq, &trampCurBand, &trampCurChan);
+        break;
+
+    case 's':
+        break;
+    }
+
+    if (trampPendingQuery == respCode)
+        trampPendingQuery = 0;
+}
+
+static bool trampIsValidResponseCode(uint8_t code)
+{
+    if (code == 'r' || code == 'v' || code == 's')
+        return true;
+    else
+        return false;
+}
+
+typedef enum {
+    S_WAIT_LEN = 0,   // Waiting for a packet len
+    S_WAIT_CODE,      // Waiting for a response code
+    S_DATA,           // Waiting for rest of the packet.
+} trampReceiveState_e;
+
+static trampReceiveState_e trampReceiveState = S_WAIT_LEN;
+static int trampReceivePos = 0;
+static uint32_t trampFrameStartUs = 0;
+
+// Frame timeout. An actual frame (16B) takes only 16.6 msec,
+// but a frame arrival may span two scheduling intervals (200msec * 2).
+// Effectively same as waiting for a next trampProcess() to run.
+
+#define TRAMP_FRAME_TIMO_US (200 * 1000)
+
+void trampReceive(uint32_t currentTimeUs)
+{
+    if (!trampSerialPort)
+        return;
+
+    if ((trampReceiveState != S_WAIT_LEN) && (currentTimeUs - trampFrameStartUs > TRAMP_FRAME_TIMO_US)) {
+        trampReceiveState = S_WAIT_LEN;
+        trampReceivePos = 0;
+    }
+
+    while (serialRxBytesWaiting(trampSerialPort)) {
+        uint8_t c = serialRead(trampSerialPort);
+        trampRespBuffer[trampReceivePos++] = c;
+
+        switch(c) {
+        case S_WAIT_LEN:
+            if (c == 0x0F) {
+                trampReceiveState = S_WAIT_CODE;
+                trampFrameStartUs = currentTimeUs;
+            }
+            break;
+
+        case S_WAIT_CODE:
+            if (trampIsValidResponseCode(c)) {
+                trampReceiveState = S_DATA;
+            } else {
+                trampReceiveState = S_WAIT_LEN;
+                trampReceivePos = 0;
+            }
+            break;
+
+        case S_DATA:
+            if (trampReceivePos == 16) {
+                uint8_t cksum = trampChecksum(trampRespBuffer);
+                if ((trampRespBuffer[14] == cksum) && (trampRespBuffer[15] == 0))
+                    trampHandleResponse();
+
+                trampReceiveState = S_WAIT_LEN;
+                trampReceivePos = 0;
+            }
+            break;
+        }
+    }
+}
+
+void trampProcess(uint32_t currentTimeUs)
+{
+    static uint32_t lastQueryRTimeUs = 0;
+
+    if (trampStatus == TRAMP_STATUS_BAD_DEVICE)
+        return;
+
+debug[0]++;
+    trampReceive(currentTimeUs);
+
+    if (trampStatus == TRAMP_STATUS_OFFLINE) {
+        if (currentTimeUs - lastQueryRTimeUs > 1000 * 1000) {
+            trampQueryR();
+            lastQueryRTimeUs = currentTimeUs;
+        }
+    } else if (trampPendingQuery) {
```

Comment:
This will resend command immediately on next call, ignoring `trampReceive` state